File: notification_hub_backend_service/src/flutter/push_demo/ios/Runner/NotificationRegistrationService.swift
//
//  NotificationRegistrationService.swift
//  Runner
//
//  Created by Mike Parker on 07/07/2020.
//

import Foundation

class NotificationRegistrationService {
    
    let NOTIFICATION_REGISTRATION_CHANNEL = "com.mobcat.pushdemo/notificationregistration"
    let REFRESH_REGISTRATION = "refreshRegistration"
    
    private let notificationRegistrationChannel : FlutterMethodChannel
    
    init(withBinaryMessenger binaryMessenger : FlutterBinaryMessenger) {
       notificationRegistrationChannel = FlutterMethodChannel(name: NOTIFICATION_REGISTRATION_CHANNEL, binaryMessenger: binaryMessenger)
    }
    
    func refreshRegistration() {
        notificationRegistrationChannel.invokeMethod(REFRESH_REGISTRATION, arguments: nil)
    }
    
}

File: notification_hub_backend_service/src/flutter/push_demo/android/app/src/main/kotlin/com/mobcat/pushdemo/services/NotificationRegistrationService.kt
package com.mobcat.pushdemo.services

import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugin.common.MethodChannel

class NotificationRegistrationService {

    companion object {
        const val NOTIFICATION_REGISTRATION_CHANNEL = "com.mobcat.pushdemo/notificationregistration"
        const val REFRESH_REGISTRATION = "refreshRegistration"
    }

    private var notificationRegistrationChannel : MethodChannel

    constructor(flutterEngine: FlutterEngine) {
        notificationRegistrationChannel = MethodChannel(flutterEngine.dartExecutor.binaryMessenger, NotificationRegistrationService.NOTIFICATION_REGISTRATION_CHANNEL)
    }

    fun refreshRegistration() {
        notificationRegistrationChannel.invokeMethod(REFRESH_REGISTRATION, null)
    }
}